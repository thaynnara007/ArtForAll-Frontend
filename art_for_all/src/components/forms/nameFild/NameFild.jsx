import React from "react";
import '../../../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import './nameFild.css';

class NameFild extends React.Component{

    render(){

        if(this.props.open){
            return(
                <div className="input-group mb-3 space">
                    <input type="text" className="form-control" placeholder={this.props.placeHolder} 
                    aria-label="Recipient's username" aria-describedby="basic-addon2" 
                    onChange={(event) => this.props.handleChange(event, this.props.data)}></input>
                    <div className="input-group-append">
                        <button className="btn btn-outline-info" type="button" onClick={this.props.onClick}>{this.props.buttonName}</button>
                    </div>
                </div>
            )
        }else return null;
    }
}

export default NameFild;