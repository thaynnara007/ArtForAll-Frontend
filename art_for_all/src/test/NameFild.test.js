import React from 'react';
import Jest from 'jest';
import Enzyme from './Enzyme';
import {shallow} from 'enzyme';
import NameFild from '../components/forms/nameFild/NameFild';

describe("NAME FILD -Testing with snapshots", () => {

    it('TEST01: it should render correctly' , () =>{
        const nameFild = shallow(<NameFild></NameFild>);
        expect(nameFild).toMatchSnapshot();
    })

    it('TEST02: it should render correctly with placeHolder=newPlaceHolder' , () =>{
        const nameFild = shallow(<NameFild placeHolder="newPlaceHolder"></NameFild>);
        expect(nameFild).toMatchSnapshot();
    })
})