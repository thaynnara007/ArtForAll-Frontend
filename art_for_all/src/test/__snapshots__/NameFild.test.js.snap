// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAME FILD -Testing with snapshots TEST01: it should render correctly 1`] = `""`;

exports[`NAME FILD -Testing with snapshots TEST02: it should render correctly with placeHolder=newPlaceHolder 1`] = `""`;
